contrib/epee/include/rolling_median.h

```diff
@@ -156,13 +156,29 @@ struct rolling_median_t
     free(data);
   }
 
+  void clear_old()
+  {
+    idx = 0;
+    minCt = 0;
+    maxCt = 0;
+    sz = 0;
+    int nItems = N;
+    while (nItems--)  //set up initial heap fill pattern: median,max,min,max,...
+    {
+      pos[nItems] = ((nItems + 1) / 2) * ((nItems & 1) ? -1 : 1);
+      heap[pos[nItems]] = nItems;
+    }
+  }
+
   void clear()
   {
     idx = 0;
     minCt = 0;
     maxCt = 0;
     sz = 0;
     int nItems = N;
+    pos = (int*) (data + N);
+    heap = pos + N + (N / 2); //points to middle of storage.
```

Comment:
I don't think these changes make any difference because once the addresses `pos` and `heap` point to never change throughout the lifetime of the object (i.e. all operations involving `pos` and `heap` always take the form of `pos[i]` and `heap[i]`, not `pos = ...` nor `heap = ...`).

Reply:
Yes, I see that now. Thanks.